third_party/move/move-binary-format/src/deserializer.rs

```diff
@@ -1778,3 +1874,66 @@ impl Opcodes {
         }
     }
 }
+
+impl SerializedBool {
+    fn from_u8(value: u8) -> BinaryLoaderResult<bool> {
+        match value {
+            0x1 => Ok(false),
+            0x2 => Ok(true),
+            _ => Err(PartialVMError::new(StatusCode::MALFORMED)
+                .with_message("malformed boolean".to_owned())),
+        }
+    }
+}
+
+impl SerializedOption {
+    fn from_u8(value: u8) -> BinaryLoaderResult<bool> {
+        match value {
+            0x1 => Ok(false),
+            0x2 => Ok(true),
+            _ => Err(PartialVMError::new(StatusCode::MALFORMED)
+                .with_message("malformed option".to_owned())),
+        }
+    }
+}
+
+impl SerializedAccessKind {
+    fn from_u8(value: u8) -> BinaryLoaderResult<AccessKind> {
+        use AccessKind::*;
+        match value {
+            0x1 => Ok(Reads),
```

Comment:
```suggestion
            x if x == Self::READ as u8 => Ok(Reads),
```

Perhaps this pattern is a bit less error prone and maintainble?

Reply:
As @brmataptos pointed out this is the current way how this handled for a dozen or so types, so lets stay with this. The direct match is definitely significantly faster.